### PR TITLE
fix: nats-server runs as root

### PR DESCRIPTION
**Pull request by friedrichwilken**
**Description**

Changes proposed in this pull request:

- Change user for NATS Pods to `1000` (aka non-root user).
- Add an Init-Container to change the owner of the JetStream storage to `1000`.
- For the `e2e-upgrade-test` GHAW, add logging for `StatefulSet`, `Deployment`, `Pods` and `ConfigMaps` that will always be displayed.

**Test**
We tested this with a pub-sub setup via the [eventing-tools](https://github.com/kyma-project/eventing-tools): 
1. We created a publisher (without a subscriber) that sends events to fill the file storage of the `NATS-server`. We eventually stop the publisher, so we have a defined number of events.
2. We changed the `NATS-manager` image to this PR's image; this will update the `StatefulSet` which will trigger a rolling update of the `Pods`, starting with the `InitContainer` that changes the owner and rights of the files in the `persistantStorage`.
3. We created a subscriber (without a publisher) that will receive the messages created in step 1 and stored in the file storage.
All created messages where received; no messages were lost.

We also checked the owner of the persistent storage:
Before the `NATS-manager` image change the owner was `root` and the `group` was `10001`.
After the owner was `1000` and the `group` was `10001`.

**Related issue(s)**

### CODE REVIEW

File: .github/workflows/pull-e2e-upgrade-test.yaml [hunk 1]
```diff
@@ -74,6 +74,26 @@ jobs:
...
+      - name: Show all Pods in Namespace kyma-system
+        if: ${{ always() }}
+        run: |
+          kubectl get po -n kyma-system -o yaml
+
+      - name: Show all ConfigMaps in Namespace kyma-system
+        if: ${{ always() }}
+        run: |
+          kubectl get cm -n kyma-system -o yaml
+
```
Comment (marcobebway) [inline .github/workflows/pull-e2e-upgrade-test.yaml:96]: Why is this needed? can we remove it?

File: resources/nats/templates/statefulset.yaml [hunk 2]
```diff
@@ -94,6 +94,15 @@ spec:
       # reload to the server without restarting the pod.
       shareProcessNamespace: true
 
+      initContainers:
+      - name: fix-jetstream-storage
```
Comment (marcobebway) [inline resources/nats/templates/statefulset.yaml:98]: `fix-jetstream-storage` -> `change-storage-permissions`

File: .github/workflows/pull-e2e-upgrade-test.yaml (hunk 1 above)
Comment (friedrichwilken) [reply .github/workflows/pull-e2e-upgrade-test.yaml]: I reverted the changes.

File: resources/nats/templates/statefulset.yaml (hunk 2 above)
Comment (friedrichwilken) [reply resources/nats/templates/statefulset.yaml]: Done.

**Review by marcobebway — APPROVED** [review]
